src/app/routes/getInitialData/index.js

```diff
@@ -6,19 +6,48 @@ import onClient from '#lib/utilities/onClient';
 import getBaseUrl from './utils/getBaseUrl';
 import getPreprocessorRules from './utils/getPreprocessorRules';
 
-const logger = nodeLogger(__filename);
+export const logger = nodeLogger(__filename);
 const STATUS_CODE_OK = 200;
 const STATUS_CODE_BAD_GATEWAY = 502;
 const STATUS_CODE_NOT_FOUND = 404;
 const upstreamStatusCodesToPropagate = [STATUS_CODE_OK, STATUS_CODE_NOT_FOUND];
+const validRendererEnvironments = ['int', 'test', 'live'];
 
 const ampRegex = /(.amp)$/;
 
 const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+// eslint-disable-next-line consistent-return
+export const getUrl = pathname => {
```

Comment:
may want to disable all this in production and fall back to the original implementation - this code runs on every request, and will have no effect when live

eg:
```
const basePath = `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
if (process.env.APP_ENV === 'live') return basePath;
```

or in the interests of removing complexity from this function, leave the original as it is, and switch between them in `fetchData` (line 93-ish)

Reply:
I did want to implement this i.e. disabled in production - however, if we do supply a query string, it is included in the pathname then`${baseUrl}${pathname.replace(ampRegex, '')}.json` converts to the following: `www.bbc.com/test/12345?param=test` -> `www.bbc.com/test/12345?param=test.json` which I think is incorrect

Reply:
ah good point - I believe that is the case in the current code too, but you've fixed a bug we don't even know we have yet :D

Reply:
Was just discussing this with Karina, I think we should explicitly prevent overrides from ever working in the live environment in simorgh in case downstream behaviour changes. Ideally we should strip the code from the live bundle entirely but that may be a follow-up task.

Reply:
See @graemelindsayBBC's comment
> Live Simorgh should never be able to load TEST content.

Reply:
Query string params are now removed in live environment